policyengine_core/simulations/simulation.py

```diff
@@ -1395,77 +1412,28 @@ def extract_person(
 
         return json.loads(json.dumps(situation, cls=NpEncoder))
 
-    def _get_macro_cache(
-        self,
-        variable_name: str,
-        period: str,
-    ):
+    def check_macro_cache(self, variable_name: str, period: str) -> bool:
         """
-        Get the cache location of a variable for a given period, if it exists.
+        Check if the variable is able to have cached value
         """
-        if not self.is_over_dataset:
-            return None
+        is_cache_available = True
+        if self.is_over_dataset:
```

Comment:
Since this PR started- we've added the ability to load DataFrames directly in simulations. Could we add an extra condition here that simulations from DataFrames are excluded from the macro cache? Thanks!

Reply:
@nikhilwoodruff Could you describe the syntax necessary for this?

Reply:
I take it it would involve using 
```
if self.dataset.data_format == Dataset.TIME_PERIOD_ARRAYS:
```
?

Reply:
We need to check if the data format is `Dataset.FLAT_FILE` and disable the entire macro cache if yes.